provider/execution/execution.go

```diff
@@ -397,10 +397,20 @@ func (e TableExecutor) resolveResourceValues(ctx context.Context, meta schema.Cl
 }
 
 // resolveColumns resolves each column in the table and adds them to the resource.
-func (e TableExecutor) resolveColumns(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, cols []schema.Column) diag.Diagnostics {
+func (e TableExecutor) resolveColumns(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, cols []schema.Column) (diags diag.Diagnostics) {
+	var col string
+
+	defer func() {
+		if r := recover(); r != nil {
+			stack := string(debug.Stack())
+			e.Logger.Error("resolve columns recovered from panic", "panic_msg", r, "stack", stack, "column_name", col)
+			diags = fromError(fmt.Errorf("column resolve panic: %s", r), diag.WithResourceName(e.ResourceName), diag.WithSeverity(diag.PANIC),
+				diag.WithSummary("resolve column %q in table %q recovered from panic", col, e.Table.Name), diag.WithDetails("%s", stack))
+		}
+	}()
```

Comment:
defer copies col by value, so the value is incorrect. Tried to recreate in playground

https://go.dev/play/p/8j21HYQVJ5a

Would also add  some tests here.

Reply:
https://go.dev/play/p/dqhjbzg2hx2 better example, but correct

Reply:
Test added